server/src/main/java/org/elasticsearch/common/logging/NodeAndClusterIdConverter.java

```diff
@@ -49,14 +50,14 @@ public NodeAndClusterIdConverter() {
     }
 
     /**
-     * Updates only once the clusterID and nodeId
+     * Updates only once the clusterID and nodeId.
+     * Note: Should only be called once. Subsequent executions will throw {@link org.apache.lucene.util.SetOnce.AlreadySetException}.
      *
      * @param nodeId      a nodeId received from cluster state update
      * @param clusterUUID a clusterId received from cluster state update
-     * @return true if the update was for the first time (successful) or false if for another calls (does not updates)
      */
-    public static boolean setOnce(String nodeId, String clusterUUID) {
-        return nodeAndClusterId.compareAndSet(null, formatIds(clusterUUID, nodeId));
+    public static void setOnce(String nodeId, String clusterUUID) {
```

Comment:
I still dont' like this being "setOnce", I think it should be parallel to the node name listener?

Reply:
as this is almost the same as NodeNameConverter I think you are right this method should be name in similar manner. Will rename to `setNodeIdAndClusterId`